### ISSUE TITLE
Disabling Module  should disable Content Parts defined in the module.

### ISSUE THREAD

**Issue by ns8482e** [OrchardCMS/OrchardCore#4172]
1. I created a Blank Site.
2. Enabled "Flows" Module.
3. Using Admin UI,  Created "Paragraph" Content type with stereotype widget. 
4. Created "Page" Content type with Title, Alias, Autoroute, Common and Flow Parts, with creatable, listable.
5. Created a New Content Item of content type "Page" titled "Home Page" with some content in "Paragraph", set as home page and Published 
On Front end Home Page is displayed with title and paragraph.
6. Disabled "Flows" module
7. "Paragraph" is now not visible at home page and not editable in Admin UI
However "Flow" is still visible in "Edit" Content Type definition of Page and View List Part page.

As "FlowPart" is defined in code in module - disabling module should hide "FlowPart" from content type that added the part and in  List Parts page Admin UI.

**Comment by hishamco** [comment]
Make sense, but how the content parts are disable?!!

**Comment by ns8482e** [comment]
I disabled the Flow Module 
 
![image](https://user-images.githubusercontent.com/23270244/64354555-484ca400-cfc5-11e9-8e42-26cdc4f39546.png)

But the Part is still visible in Content Type Definition 

![image](https://user-images.githubusercontent.com/23270244/64354685-877af500-cfc5-11e9-9105-c0316c28534d.png)

And Part List Page

![image](https://user-images.githubusercontent.com/23270244/64354820-c0b36500-cfc5-11e9-8698-75aa9cfb814a.png)

It's confusing for user if "Part" is visible in definition but can't participate in View or Edit.

**Comment by hishamco** [comment]
Probably showing a notification for a use is a good option

**Comment by sebastienros** [comment]
I think the content definitions should be built from different sources. One source is the current document that is managed in the UI, and another one would be by code, such that disabling the feature would hide them. As of today, disabling a feature is not sufficient as it alters the dynamic metadata.
We should also have a flag in the part to note that it's a code part (not a metadata one) and it might appear differently in the UI, like not being able to delete it at least.

**Comment by sebastienros** [comment]
Another alternative is to implement the uninstall event and remove the parts that the feature has added.

**Comment by ns8482e** [comment]
Hide or Disabled is non destructive way  compared to removing it from Parts definition.  It would allow Modules/Features can be used in scenarios like subscription based where Features can disabled at the end of subscription that can be re-enabled when the subscription re-activated again at later stage - without affecting Content Type definition or Data

**Comment by hishamco** [comment]
> Hide or Disabled is non destructive way compared to removing it from Parts definition

You are right, but it depends on the scenario,  while you added the parts from the UI it should be removed, but if they added from code they will be created next time you enable the module

I'm working on a PR for this I will see what can I come up with ..

**Comment by ns8482e** [comment]
Assume that for above mentioned scenario, "Flow" module is subscription based module that gets disabled at the end of subscription,

Now Site owner subscribes to Flow module, and creates "Page" content type using Admin UI and starts using the "FlowPart" and create some 10-15 pages.

Now at the end of subscription, "Flow" module gets disabled along with all its View/Edit functionality. 
However JSON data of existing pages created with "FlowPart" should remain unaffected. And The FlowPart  in existing pages can't be edited/viewed or no new page can be created with "FlowPart".

When the "Flow" modules gets enabled again when re-subscribed, the FlowPart of exiting Pages will become available for View/Edit and new Pages with FlowPart can be created.

This doesn't require Site Owner to change  Content Type definition of Page every time the Module gets enabled/disabled.

**Comment by sebastienros** [comment]
Note that by keeping the parts around we don't have to delete the content items using these
